ical/checks.go

```diff
@@ -1,123 +1,132 @@
 package ical
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/Fesaa/ical-merger/config"
 	"github.com/Fesaa/ical-merger/log"
 	ics "github.com/arran4/golang-ical"
 )
 
-func (ical *LoadediCal) Check(event *ics.VEvent) bool {
-	s := ical.source
-	if s.Rules == nil || len(s.Rules) == 0 {
+func (c *LoadediCal) Check(event *ics.VEvent) bool {
+	if len(c.source.Rules) == 0 {
 		return true
 	}
-	for _, rule := range s.Rules {
-		if ical.apply(&rule, event) {
+	for _, rule := range c.source.Rules {
+		if c.apply(&rule, event) {
 			return true
 		}
 	}
 	return false
 }
 
-var checks = map[string]func(r *config.Rule, input string) bool{
-	"CONTAINS": func(r *config.Rule, input string) bool {
-		input = r.Transform(input)
-		for _, s := range r.Data {
-			s = r.Transform(s)
-			if strings.Contains(input, s) {
-				return true
-			}
-		}
-		return false
-	},
-	"NOT_CONTAINS": func(r *config.Rule, input string) bool {
-		input = r.Transform(input)
-		for _, s := range r.Data {
-			s = r.Transform(s)
-			if strings.Contains(input, s) {
-				return false
-			}
-		}
-		return true
-	},
-	"EQUALS": func(r *config.Rule, input string) bool {
-		input = r.Transform(input)
-		for _, s := range r.Data {
-			s = r.Transform(s)
-			if input == s {
-				return true
-			}
-		}
-		return false
-	},
-	"NOT_EQUALS": func(r *config.Rule, input string) bool {
-		input = r.Transform(input)
-		for _, s := range r.Data {
-			s = r.Transform(s)
-			if input == s {
-				return false
-			}
-		}
-		return true
-	},
+func (c *LoadediCal) apply(r *config.Rule, event *ics.VEvent) bool {
+	switch r.Check {
+	// Filters
+	case filterContainsTerm:
+		return c.filterContains(r, event)
+	case filterNotContainsTerm:
+		return c.filterNotContains(r, event)
+	case filterEqualsTerm:
+		return c.filterEquals(r, event)
+	case filterNotEqualsTerm:
+		return c.filterNotEquals(r, event)
+
+	// Modifiers
+	case modifierFirstOfDayTerm:
+		return c.modifierFirstOfDay(event)
+	case modifierFirstOfMonthTerm:
+		return c.modifierFirstOfMonth(event)
+	case modifierFirstOfYearTerm:
+		return c.modifierFirstOfYear(event)
+	default:
+	}
+	log.Logger.Warn("Check not found", "rule_name", r.Name, "check", r.Check)
+	return false
 }
 
-var special_checks = map[string]func(ical *LoadediCal, event *ics.VEvent) (*bool, error){
-	"FIRST_OF_DAY": func(ical *LoadediCal, event *ics.VEvent) (*bool, error) {
-		start, e := event.GetStartAt()
-		if e != nil {
-			return nil, e
-		}
+const (
```

Comment:
Can you move (or copy) all documentation to the constants, and expose them? So they're more accessible in the godoc.